contrib/mesos/pkg/minion/server.go

```diff
@@ -258,15 +215,54 @@ func (ms *MinionServer) Run(hks hyperkube.Interface, _ []string) error {
 	ms.clientConfig = clientConfig
 
 	// run subprocesses until ms.done is closed on return of this function
-	defer close(ms.done)
 	if ms.runProxy {
-		go runtime.Until(ms.launchProxyServer, 5*time.Second, ms.done)
+		ms.launchProxyServer()
 	}
-	go runtime.Until(ms.launchExecutorServer, 5*time.Second, ms.done)
 
-	// wait until minion exit is requested
-	// don't close ms.exit here to avoid panics of go routines writing an error to it
-	return <-ms.exit
+	// abort closes when the kubelet-executor dies
+	abort := ms.launchExecutorServer()
+	shouldQuit := termSignalListener(abort)
+	te := mergeTaskOutput(ms.tasks, shouldQuit)
+
+	// TODO(jdef) do something fun here, such as reporting task completion to the apiserver
```

Comment:
came in handy for unit testing (which we didn't have at all before). this todo was just a random idea that occurred to me, something that I thought might be useful for operators at some point; there's no urgency for it.

i'm pretty hesitant to eliminate the events stuff since I think it helps testability and establishes a foundation for (possibly) reporting component failure events/metrics. if it's too complex, I'm open to other ideas. the prior state of process management didn't quite compare to what s6 was offering us before. this comes much closer.